org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/breakpoints/ValidBreakpointLocationLocator.java

```diff
@@ -1066,10 +1066,16 @@ public boolean visit(LambdaExpression node) {
 
 						}
 					}
+
+				// Lambda body can be a block (which is handled above) or an (arbitrary) expression. So maybe the
+				// following cases are insufficient and should be replaced by handling the general
+				// org.eclipse.jdt.core.dom.Expression.
 				} else if (body instanceof LambdaExpression) {
 					body.accept(this);
 				} else if (body instanceof MethodInvocation) {
 					body.accept(this);
+				} else if (body instanceof ClassInstanceCreation) {
```

Comment:
since the handling is the same may be use OR in the condition or even a general "instanceof Expression"